qubesbuilder/plugins/installer/__init__.py

```diff
@@ -51,8 +52,13 @@ class InstallerPlugin(DistributionPlugin):
     InstallerPlugin creates Qubes OS ISO
     """
 
+    name = "installer"
     stages = ["init-cache", "prep", "build", "sign", "upload"]
-    dependencies = ["chroot_rpm"]
+    dependencies = [
+        PluginDependency("chroot_rpm"),
+        ComponentDependency("qubes-release"),
+        ComponentDependency("repo-templates"),
```

Comment:
Can't you use repo definitions extracted from kickstart file already? `ksparser` extracts it to separate conf already, and keys are there already too. This way, you could avoid extra dependency here, and template repos would be defined for installer in a single place.

But if the dependency is to stay, it needs to be added in config for tests, as currently it fails.

Reply:
I can, that was the original idea. But, you might want to cache much more templates than ones into kickstart file only.

Reply:
Those are being used for installer only, right? And if you want more, you already can provide custom kickstart file (possibly with more repos, and/or more templates listed).
Also, having repos from kickstart file already gives you an option to enable just selected repos, so the enablerepo logic is not necessary then.

Reply:
Yes, I finally went with the approach of using kickstart only but you can keep config definition of which templates you want to have a cache.